File: python/PerformanceTest.py
from FlatGrid import FlatGrid
from NestedGrid import NestedGrid
from DictGrid import DictGrid
from FlatDictGrid import FlatDictGrid
from NestedNumpyGrid import NestedNumpyGrid as npNestedGrid
from Curve import Curve
from time import process_time

def test_performance(template, grid, curve):
	template.radiate_value_around_position(template.center, template.center[0], curve)
	start = process_time()
	for i in range(grid.width):
		for j in range(grid.height):
			grid.add_grid_at_pos(template, (i, j))
	duration = process_time() - start
	print(duration)

curve = Curve(-1.0, 1.0, 0.0, 1.0)
print("NestedGrid")
for i in range(30):
	test_performance(NestedGrid(9, 9), NestedGrid(600, 600), curve)
print("FlatGrid")
for i in range(30):
	test_performance(FlatGrid(9, 9), FlatGrid(600, 600), curve)
print("DictGrid")
for i in range(30):
	test_performance(DictGrid(9, 9), DictGrid(600, 600), curve)
print("FlatDictGrid")
for i in range(30):
	test_performance(FlatDictGrid(9, 9), FlatDictGrid(600, 600), curve)
print("npNestedGrid")
for i in range(30):
	test_performance(npNestedGrid(9, 9), npNestedGrid(600, 600), curve)

File: cpp/PerformanceTest.cpp
#include "Curve.h"
#include "FlatGrid.h"
#include "NestedGrid.h"
#include <chrono>
#include <iostream>

template <typename GridType>
void performance_test(int size_templ, int size_grid, const Curve &curve) {
	GridType template_grid(size_templ, size_templ);
	GridType grid(size_grid, size_grid);
	template_grid.radiate_value_around_position(template_grid.get_center(), template_grid.get_center().x, curve, 1.0f);
	auto start = std::chrono::high_resolution_clock::now();
	for (int i = 0; i < grid.get_width(); i++) {
		for (int j = 0; j < grid.get_height(); j++) {
			grid.add_grid_at_pos(template_grid, Vector2i(i, j));
		}
	}
	auto end = std::chrono::high_resolution_clock::now();
	auto duration = std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
	std::cout << duration / 1000000.0 << std::endl;
}

int main() {
	const Curve testcurve = { -1.0f, 1.0f, 0.0f, 1.0f };
	
	std::cout << "NestedGrid" << std::endl;
	for (int i = 0; i < 30; i++) {
		performance_test<NestedGrid>(9, 600, testcurve);
	}

	std::cout << "FlatGrid" << std::endl;
	for (int i = 0; i < 30; i++) {
		performance_test<FlatGrid>(9, 600, testcurve);
	}

	char waiter;
	std::cin >> waiter;

	return 0;
}